File: Easy/0191_number-of-1-bits/number-of-1-bits.py
#!/bin/python3

from typing import List

class Solution:
    def hammingWeight(self, n: int) -> int:
        count : int = 0
        while n > 0 :
            count += n & 1
            n >>= 1
        return count

def main():
    tests : List[int] = [
        11,
        128,
        2147483645,
        0,
    ]

    for n in tests:
        print(f"---\nN: {n} hammingWeight: {Solution().hammingWeight(n)}")

if __name__ == "__main__":
    main()

File: Easy/0191_number-of-1-bits/number-of-1-bits.go
package main

import (
	"fmt"
)

func hammingWeight(n int) int {
	count := 0
	for n > 0 {
		count += n & 1
		n >>= 1
	}
	return count
}

func main() {
	tests := []int{
		11,
		128,
		2147483645,
		0,
	}

	for _, n := range tests {
		fmt.Printf("---\nN: %d hammingWeight: %d\n", n, hammingWeight(n))
	}
}
